code/frameworks/angular/src/client/docs/compodoc.ts

```diff
@@ -223,8 +226,11 @@ const resolveTypealias = (compodocType: string): string => {
 
 export const extractArgTypesFromData = (componentData: Class | Directive | Injectable | Pipe) => {
   const sectionToItems: Record<string, ArgType[]> = {};
+  const componentClasses = FEATURES.angularFilterNonInputControls
```

Comment:
Don't you think we should also add the output ?

IMHO: In the vast majority of cases the "user interface" of an Angular component is this input/output. 

 I would change the default list to `'inputsClass', 'outputsClass'`  and I will add a flag 
`angularCompodocAddAllSection: boolean` (something like that) 😁 
or `angularCompodocSection: string[]` (and let the user choose if he wants more than input / output)

Reply:
I like the idea of choosing, maybe with regex.

If this is only going to affect what properties have controls or get recognized as outputs for actions then it isn't a big problem having a boolean for just inputs/outputs. Often a story is focused on only a few inputs, so I wouldn't want all possible inputs in the code display. I remember it used to just show a button for properties that could have a control, but didn't have a value set, and clicking it would then add that control.

If it affects what is displayed in the ArgsTable, not just which have controls, then it is more important to have something like regex filtering. I like everything public to be shown, with the type and doc comment. Since the template can only access public properties, I prefix properties that should be private with a `_` to indicate that it should be treated as private even though it has to be public for the template to access it.

Reply:
@ThibaudAV but your Angular outputs are only relevant for your Actions tab right? Not Controls?

Reply:
@JamesHenry I could be wrong, but I don't think @ThibaudAV is asking about including outputs for the Controls tab necessarily.

In my opinion the word "Controls" is over used and causes confusion when discussing ArgTypes. The "Controls" tab displays an ArgsTable, but the most used feature of the ArgsTable is probably it's Controls column. The "Controls" tab and "Actions" tab are just for display and user interaction.

Underneath, the code is deciding what properties to handle by looking at the `args`. If you look a few lines below this, the properties will be iterated and outputs will be defined as actions.

@shilman @ThibaudAV If my above explanation is accurate, do you think `angularFilterNonInputControls` could be a misleading name. Unless I am overlooking something, this is actually an ArgsTable filter. I don't know if ArgsTable is a recognizable name for users unfamiliar with the internal code though.

Reply:
@Marklb I'm not 100% what you are advocating for. What value does rendering Angular Outputs in the ArgsTable in Controls have?

Real world example: https://gemini.opensesame.com/latest/?path=/story/components-accordion--accordion

<img width="1296" alt="image" src="https://user-images.githubusercontent.com/900523/203772834-83873a94-9f99-4ab9-a0cd-b0052ed2da8b.png">

All that real estate for a name and a dash that I cannot do anything with? They do not control the Component in any way, so rendering them under Controls (even above Inputs it seems?) is very unintuitive.

When it comes to Angular Outputs what I care about is the Actions tab of that same Component:

<img width="1301" alt="image" src="https://user-images.githubusercontent.com/900523/203772894-6d009e4a-0456-4b3d-ac63-ccc16a7dd833.png">

If you had to render a list of available Outputs anywhere, then the top of the Actions tab would make a lot more sense IMO

Reply:
@JamesHenry I think your point makes sense, but that is probably something that could be done as a change to extend the configurability of the Controls panel, from the docs addon. This PR seems to be filtering information that is not used by only controls, though. I don't think outputs would automatically show up in Actions if filtered here.

As for the value of Angular Outputs in the ArgsTable, I stay on the "Canvas" majority of the time I am using Storybook. That table is my overview of what the properties and methods are that I may interact with in Storybook as Controls or in my application.

When I am using a monitor wide enough to easily fit more columns, I would prefer the expanded ArgsTable that is used when there is not any Controls and when on the "Docs" view. The following is what the expanded ArgsTable looks like on "Docs".

![Docs](https://user-images.githubusercontent.com/8348647/203953859-e4a359ce-51f3-4b7d-8ee3-39827d7c3608.PNG)

The Control column on the OUTPUTS and METHODS groups rows should not have a Control, but the point I was making is with that table I can see what is available and interact with the Controls from a single spot.

For the outputs, I can easily see what outputs are available and the Description column shows me the JSDoc comment for that property and the property type. *It should be showing the type as `EventEmitter<DoneEvent>`, instead of just `EventEmitter`.*

For the inputs, the Control column only gives an appropriate control for the value. By having the Description and Default columns, I can read the JSDoc comment describing what the input actually is for, I can see the actual type of the property, and a reminder what the default value was if I have been adjusting the controls.

It is rare that you would have properties intended to be changed that you wouldn't also want as an input, but there are situations it makes sense.

The methods shouldn't have a Control, but often a Component or Directive are intended to be used outside of the template. So it is nice to have the methods that are intended to be used, in the same spot as the properties that are intended to be used, along with the JSDoc comment and return type.

The Controls panel looks much more clean without the additional columns, but the way I am typically using Storybook can easily fit the extra columns and avoid having to click back and forth between "Canvas" and "Docs".

The real world example you sent is not using the ArgsTable for their "Docs" page and I agree that the non-input rows are not very useful when the ArgsTable isn't expanded.

I agree that outputs should not be above the inputs. It does seem like inputs should be the top group though. I thought the ArgsTable supported custom sorting for the groups, but I am only seeing a could options for row sorting.